### ISSUE TITLE
[Nette 3.1] Add upgrade set

### ISSUE THREAD

**Issue by TomasVotruba** [rectorphp/rector#5142]
See https://forum.nette.org/cs/34080-novinky-v-nette-3-1-je-venku

**Comment by lulco** [comment]
I can look at this and prepare nette_31 set if it is not top urgent priority :)

**Comment by TomasVotruba** [comment]
That's would be great. We might upgrade one private project next week, but I'm not sure when Nette 3.1 will come up, as there are other sets with higher priority.

Let me know when you start :)

**Comment by lulco** [comment]
I just read the article and look at some diffs. It looks like there are only some class renamings (with back compatibility in aliases), few parameter changes. So it would be easy (I hope).

But there are many config (neon) changes (Like it was in 3.0 too). Do you have any idea how we can do also these neon changes? In my projects I do just simple regex replace, but there can be false positives, so I would like to use some sophisticated way. I was trying to use Neon::decode -> Neon::encode but it completely changed my config files :(
Also some changes are based on extension schema and Neon itself have no idea about it (like your db password has to be string so you have to change 123 to '123')

**Comment by TomasVotruba** [comment]
You've hit nail of the format preserving :) the NEON and YAML don't have those. That's why we've switched configs to PHP.

We used regular expressions as well, but it was so un-reliable and cause more harm than work done.
One way to fix this is creating a format preserving printer for NEON. 
php-parser has one for PHP. I'm working on one for docblocks here https://github.com/rectorphp/phpdoc-parser-printer.
